src/internal.c

```diff
@@ -5950,6 +5962,48 @@ int wolfSSL_CTX_IsPrivatePkSet(WOLFSSL_CTX* ctx)
 }
 #endif /* HAVE_PK_CALLBACKS */
 
+static void InitSuites_EitherSide(Suites* suites, ProtocolVersion pv, int keySz,
+        word16 haveRSA, word16 havePSK, word16 haveDH, word16 haveECDSAsig,
+        word16 haveECC, word16 haveStaticECC,
+        word16 haveFalconSig, word16 haveDilithiumSig, word16 haveAnon,
+        int side)
+{
+    /* make sure server has DH parms, and add PSK if there */
+    if (side == WOLFSSL_SERVER_END) {
+        InitSuites(suites, pv, keySz, haveRSA, havePSK, haveDH, haveECDSAsig,
```

Comment:
Set haveDH to TRUE when side is WOLFSSL_SERVER_END

Reply:
It was set to `ssl->options.haveDH` before this PR. Is there any specific reason to change it in this PR?

Reply:
Simplify code.

Reply:
Before this PR the code looked like this:

```
    /* make sure server has DH parms, and add PSK if there */
    if (ssl->options.side == WOLFSSL_SERVER_END) {
        InitSuites(ssl->suites, ssl->version, keySz, haveRSA, havePSK,
                   ssl->options.haveDH, ssl->options.haveECDSAsig,
                   ssl->options.haveECC, TRUE, ssl->options.haveStaticECC,
                   ssl->options.haveFalconSig, ssl->options.haveDilithiumSig,
                   ssl->options.haveAnon, TRUE, ssl->options.side);
    }
    else {
        InitSuites(ssl->suites, ssl->version, keySz, haveRSA, havePSK, TRUE,
                   ssl->options.haveECDSAsig, ssl->options.haveECC, TRUE,
                   ssl->options.haveStaticECC, ssl->options.haveFalconSig,
                   ssl->options.haveDilithiumSig, ssl->options.haveAnon, TRUE,
                   ssl->options.side);
    }
```

Now it is 

```
    /* make sure server has DH parms, and add PSK if there */
    if (side == WOLFSSL_SERVER_END) {
        InitSuites(suites, pv, keySz, haveRSA, havePSK, haveDH, haveECDSAsig,
                   haveECC, TRUE, haveStaticECC, haveFalconSig,
                   haveDilithiumSig, haveAnon, TRUE, side);
    }
    else {
        InitSuites(suites, pv, keySz, haveRSA, havePSK, TRUE, haveECDSAsig,
                   haveECC, TRUE, haveStaticECC, haveFalconSig,
                   haveDilithiumSig, haveAnon, TRUE, side);
    }
```

It should be `TRUE` when client side. Do you want me to change the behaviour in this PR?